### ISSUE TITLE
build from source on opensuse 42.1 failed

### ISSUE THREAD

**Issue by gecko-dev** [SoftEtherVPN/SoftEtherVPN#214]
Issuse: hamcorebuilder terminated with Segmentation fault.

source: softether-src-v4.18-9570-rtm.tar.gz
### 
```
> Core was generated by `tmp/hamcorebuilder src/bin/hamcore/ src/bin/BuiltHamcoreFiles/unix/hamcore.se2'.
> Program terminated with signal SIGSEGV, Segmentation fault.
> # 0  memset () at ../sysdeps/x86_64/memset.S:112
> 
> 112     ../sysdeps/x86_64/memset.S: Datei oder Verzeichnis nicht gefunden.
> (gdb) bt
> # 0  memset () at ../sysdeps/x86_64/memset.S:112
> # 1  0x00007f80f3bc67f8 in memset (__len=18446744073709551608, __ch=0, __dest=<optimized out>) at
> 
> /usr/include/bits/string3.h:84
> # 2  BN_lshift (r=r@entry=0x175d3e8, a=a@entry=0x175d3e8, n=n@entry=-84) at bn_shift.c:162
> # 3  0x00007f80f3bedbc6 in rsa_builtin_keygen (cb=0x7fffe8480cd0, e_value=0x175d7a0, bits=0, rsa=0x175d650) at
> 
> rsa_gen.c:437
> # 4  RSA_generate_key_ex (rsa=rsa@entry=0x175d650, bits=bits@entry=32, e_value=e_value@entry=0x175d7a0,
> 
> cb=cb@entry=0x7fffe8480cd0) at rsa_gen.c:166
> # 5  0x00007f80f3bf1ae6 in RSA_generate_key (bits=32, e_value=65537, callback=0x0, cb_arg=0x0) at rsa_depr.c:92
> # 6  0x000000000040ffc7 in RsaCheck () at src/Mayaqua/Encrypt.c:2753
> # 7  0x000000000040ff20 in RsaCheckEx () at src/Mayaqua/Encrypt.c:2730
> # 8  0x0000000000425023 in InitMayaqua (memcheck=0, debug=0, argc=3, argv=0x7fffe8481d08) at src/Mayaqua/Mayaqua.c:579
> # 9  0x00000000005a1850 in main (argc=3, argv=0x7fffe8481d08) at src/hamcorebuilder/hamcorebuilder.c:139
```

**Comment by moatazelmasry2** [comment]
Hi @gecko-dev . How much RAM do you have?

**Comment by chipitsine** [comment]
I confirm issue still exists in master branch

(I forked repo to gitlab set up gitlab-ci there)

https://gitlab.com/chipitsine/SoftEtherVPN/-/jobs/84677567

**Comment by chipitsine** [comment]
but it is not reproduced when I install opensuse 42.1 on vm myself

**Comment by chipitsine** [comment]
opensuse 42.1 is shipped with cmake 3.2.X, while currently 3.4.3 is required.

I'm going to close this bug (I was not able to reproduce on hyper-v vm with 42.1, updated cmake, 2Gb of ram and turned off swap)
